api/src/main/java/io/serverlessworkflow/api/WorkflowReader.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2020-Present The Serverless Workflow Specification Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.serverlessworkflow.api;
+
+import io.serverlessworkflow.api.types.Workflow;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Reader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+public class WorkflowReader {
+
+  public static Workflow readWorkflow(InputStream input, WorkflowFormat format) throws IOException {
+    return format.mapper().readValue(input, Workflow.class);
+  }
+
+  public static Workflow readWorkflow(Reader input, WorkflowFormat format) throws IOException {
+    return format.mapper().readValue(input, Workflow.class);
+  }
+
+  public static Workflow readWorkflow(Path path, WorkflowFormat format) throws IOException {
+    return format.mapper().readValue(Files.readAllBytes(path), Workflow.class);
+  }
+
+  public static Workflow readWorkflowFromClasspath(String classpath) throws IOException {
```

Comment:
nit; why not just `readWorkflow` since the params are different and the argument already declares the intention?

Reply:
I added the suffix in case we add a readWorkflowFromFileName (which will should also accept a string with the file name) or a readWorkflowFromString (which will accept a string with the whole content). I do not want to add them, but if in future we need, they will need the suffix because the parameters will be string